drivers/timer/mcux_os_timer.c

```diff
@@ -111,17 +117,17 @@ static uint32_t mcux_lpc_ostick_set_counter_timeout(int32_t curr_timeout)
 			}
 		}
 
+		/* Counter is set to wakeup the system after the requested time */
+		if (counter_start(counter_dev) != 0) {
+			ret = 1;
+		}
+		counter_running = true;
```

Comment:
if counter_start fails, shouldn't counter_running be set to false ?

Reply:
Yes, this should be skipped on a fail case and the function should return. I have updated the code. Thanks for catching this.